U-Boot: add option to use a custom U-Boot binary image format

Signed-off-by: Eric Jarrige <eric.jarrige@armadeus.org>
---
--- buildroot/boot/uboot/Config.in
+++ buildroot/boot/uboot/Config.in
@@ -118,8 +118,23 @@ config BR2_TARGET_UBOOT_FORMAT_LDR
 	depends on BR2_bfin
 	bool "u-boot.ldr"
 
+config BR2_TARGET_UBOOT_FORMAT_CUSTOM
+	bool "custom binary format"
+	help
+	  Use CPU-specific binary format.
+
+	  Select this option and specify the make target in "U-Boot
+	  binary target name".
+
 endchoice
 
+config BR2_TARGET_UBOOT_FORMAT_CUSTOM_NAME
+	string "U-Boot binary target name"
+	depends on BR2_TARGET_UBOOT_FORMAT_CUSTOM
+	help
+	  Specify the kernel make target to build the kernel that you
+	  need.
+
 config BR2_TARGET_UBOOT_OMAP_IFT
 	depends on BR2_TARGET_UBOOT_FORMAT_BIN
 	depends on BR2_arm || BR2_armeb
--- buildroot/boot/uboot/uboot.mk
+++ buildroot/boot/uboot/uboot.mk
@@ -37,6 +37,9 @@ else ifeq ($(BR2_TARGET_UBOOT_FORMAT_NAN
 UBOOT_BIN          = u-boot-nand.bin
 else ifeq ($(BR2_TARGET_UBOOT_FORMAT_IMG),y)
 UBOOT_BIN          = u-boot.img
+else ifeq ($(BR2_TARGET_UBOOT_FORMAT_CUSTOM),y)
+UBOOT_BIN          = $(call qstrip,$(BR2_TARGET_UBOOT_FORMAT_CUSTOM_NAME))
+UBOOT_MAKE_TARGET  = $(UBOOT_BIN)
 else
 UBOOT_BIN          = u-boot.bin
 UBOOT_BIN_IFT      = $(UBOOT_BIN).ift

